src/Infrastructure/ViewRulesCompiler.js

```diff
@@ -0,0 +1,60 @@
+import IfcColor from './IfcColor'
+import IfcCustomViewSettings from './IfcCustomViewSettings'
+import {IFCPROPERTYSET, IFCRELDEFINESBYPROPERTIES} from 'web-ifc'
+import {interpolateColors, parseColor} from './ColorHelperFunctions'
+
+
+/* eslint-disable new-cap */
+/* eslint-disable no-magic-numbers */
+/**
+ * Convert view rules into view settings object
+ *
+ * @param {object} api
+ * @param {number} modelID
+ * @param {number} rules
+ * @return {IfcCustomViewSettings} the view settings
+ */
+export async function compileViewRules(api, modelID, rules) {
+  // Apply desired logic on model before loading
+  const propRelLines = await api.GetLineIDsWithType(modelID, IFCRELDEFINESBYPROPERTIES)
+  const allPropObjects = []
+  for (let i = 0; i < propRelLines.size(); i++) {
+    allPropObjects.push( await api.GetLine(modelID, propRelLines.get(i), true))
+  }
+  const psetObjects = allPropObjects.filter((x) => x.RelatingPropertyDefinition.type === IFCPROPERTYSET)
+  // Get only sets containing PSet_vyzn.Verlust
+  const objectsAndPropVal = psetObjects.map((a) =>
+    ({o: a.RelatedObjects[0]?.expressID,
+      p: a.RelatingPropertyDefinition.HasProperties?.find(
+          (s) => s.Name.value === 'Verlust' ||
```

Comment:
I wonder if these ought to be passed by URL param and have the application be stateless..

- on load, look for StateToken encoding param names to be colorized, and a color map
- recolor mesh in scene (unload and reload mesh if need be?)
- no state kept

Reply:
Hi @pablo-mayrgundter The current 'ViewRulesCompiler' file is a just dummy implementation that mimics the compilation of a bunch of rules against the IFC model on loading, and returns a result of 'IfcCustomViewSettings' to be used for overriding model colors.

The compiler functionality itself, and how the 'ViewRules' are stored and shared or changed will be explored in details in #475, that way it would be more generic and flexible, but for now it's fixed with the predefined view rules in #603

They're both big topics (compiling and storing/changing the ViewRules) that this PR will get too complicated if it tried to cover both of them as well

this PR's main focus was validating the workflow of having a predefined set of rules, that would be compiled and applied on model load, that's why there's not much focus on the actual compilation of viewRules itself